discovery_test.go

```diff
@@ -291,18 +302,3 @@ func TestGossipSubDiscoveryAfterBootstrap(t *testing.T) {
 		}
 	}
 }
-
-func waitUntilGossipsubMeshCount(ps *PubSub, topic string, count int) {
```

Comment:
can't we tell staticchek somehow to not barf here? I don't like having it inline.

Reply:
This is indeed a not very helpful staticcheck message. It says the function is unused, apparently because the test is skipped.

Smells like a false positive to me: https://github.com/dominikh/go-tools/issues/968.

I'll just add a `nolint` comment here, I agree that inlining is not a nice solution.

Reply:
Done.